docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,238 @@
+New search API
+==============
+
+Goals
+-----
+
+- Allow to configure search at the API level,
+  instead of having the options in the database.
+- Allow to search a group of projects/versions at the same time.
```

Comment:
Is one of the goals to have the same search UX across the dashboard & in-doc search? That is one of my goals :D

Reply:
I didn't put it as a goal, but there are some questions related to this a the end of the document.

With same UX, do you mean to match the API search what we do in the dashboard or the other way around? And also, would this be using the same API or is it okay to handle that within the global search view?

Matching the UX from the API in the dashboard and using its own view would be easier to integrate, we will lose some of the features of the global search (like the filters that show the number of results), but as mentioned in the document, I'm not sure if that's a useful feature.

Reply:
Also, we would need to change the projects search as you noted in https://github.com/readthedocs/readthedocs.org/pull/9533#discussion_r954321748.

Reply:
Yea, I'm thinking of the dashboard UX matching the existing one with our live-updated doc search. It would be the same UI/UX, but default to different filtering. In-doc search would default to searching the project, whereas global search would default to global file search, unless you were on the project dashboard, then it defaults to project? 

I think having good search that works consistently is more important than the filters. I think we can add the filter UI in if we have requests for it from users.

Reply:
Agreed. I like the idea of having the same search UX everywhere.